src/OrleansRuntime/Services/GrainService.cs

```diff
@@ -40,16 +41,16 @@ protected GrainServiceStatus Status
         protected IGrainServiceConfiguration Config { get; private set; }
 
         /// <summary>Only to make Reflection happy</summary>
-        protected GrainService() : base(null, null)
+        protected GrainService() : base(null, null, null)
         {
             throw new Exception("This should not be constructed by client code.");
         }
 
         /// <summary>Constructor to use for grain services</summary>
-        protected GrainService(IGrainIdentity grainId, Silo silo, IGrainServiceConfiguration config) : base((GrainId)grainId, silo.SiloAddress, lowPriority: true)
+        protected GrainService(IGrainIdentity grainId, Silo silo, IGrainServiceConfiguration config, ILoggerFactory loggerFactory) : base((GrainId)grainId, silo.SiloAddress, lowPriority: true, loggerFactory:loggerFactory)
         {
             typeName = this.GetType().FullName;
-            Logger = LogManager.GetLogger(typeName);
+            Logger = new LoggerWrapper<GrainService>(loggerFactory);
```

Comment:
This is a Logger that shares with the derived types, and it was using the actual type name before, and I think that should be preserved

Reply:
good catch